### PR TITLE
Update for Sorting Query Terms

### PR DESCRIPTION
**Pull request by apmoriarty**
When sorting a query by implied or actual cardinality, now perform a secondary alphabetical sort within each 'bucket'. 

Updated expectations for unit tests that were sensitive to term ordering.

### CODE REVIEW

**Review by jwomeara — COMMENTED** [review]
looks safe from a query microservices perspective.  happy to approve from a codeowner perspective once this has 2 approvals.

**Review by ivakegg — APPROVED** [review]

**Review by alerman — APPROVED** [review]

**Pull request merged**
